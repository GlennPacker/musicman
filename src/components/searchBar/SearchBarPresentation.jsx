import { Box, Button, Container, TextField } from '@material-ui/core';
import React, { useState } from 'react';
import PropTypes from 'prop-types';

const SearchBarPresentation = ({ search }) => {
  const [form, setForm] = useState({
    searchTerm: ''
  });
  
  const handleChange = e => {
    setForm({
      searchTerm: e.target.value
    });
  };

  const submitSearch =  (e) => {
    e.preventDefault();
    search(form);
  };

  return (
    <Container>
      <form 
        noValidate 
        autoComplete="off"
        onSubmit={ submitSearch }
      >
        <Box 
          display="flex"
          flexDirection="row"
          justifyContent="flex-end"
        >
          <TextField 
            id="searchTerm"
            name="searchTerm"
            label="Search"
            fullWidth
            value={ form.searchTerm }
            onChange={ handleChange }
          />
          <Button 
            variant="contained" 
            color="primary"
            type="submit"
            onClick={ submitSearch }
          >
            Search
          </Button>
        </Box>
      </form>
    </Container>
  );
};

SearchBarPresentation.propTypes = {
  search: PropTypes.func.isRequired
};

export default SearchBarPresentation;