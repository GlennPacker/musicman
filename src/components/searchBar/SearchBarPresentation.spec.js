import SearchBarPresentation from './SearchBarPresentation';
import React from 'react';
import render from 'react-test-renderer';
import { mount } from 'enzyme';

describe('SearchBarPresentation', () => {
  test('should render', () => {
    const component = render.create(
      <SearchBarPresentation 
        search={ () => {} }
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should be able to submit a valid search', () => {
    const mockSearchFunc = jest.fn();
    const wrapper = mount(<SearchBarPresentation search={ mockSearchFunc } />);
    const input = wrapper.find('input');
    const button = wrapper.find('button');

    input.simulate('change', {target: { value: 'abc' }});
    button.simulate('click');

    expect(mockSearchFunc.mock.calls.length).toBe(1);
    expect(mockSearchFunc.mock.calls[0][0].searchTerm).toBe('abc');
  });
});