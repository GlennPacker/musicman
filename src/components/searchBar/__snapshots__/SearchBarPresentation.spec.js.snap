// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBarPresentation should render 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <form
    autoComplete="off"
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="MuiBox-root MuiBox-root-1"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink={false}
          htmlFor="searchTerm"
          id="searchTerm-label"
        >
          Search
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="searchTerm"
            name="searchTerm"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Search
        </span>
      </button>
    </div>
  </form>
</div>
`;
